### ISSUE TITLE
[Tanks] Mitigation Combos

### ISSUE THREAD

**Issue by WesBosch** [PunishXIV/WrathCombo#46]
(There was no Suggestion tag, so hopefully Enhancement is ok)

I'd like to propose an option to combine personal mits and party mits to one button each, resulting in having 1 button for personal mits, and 1 button for party mits. Mechanically this is similar to how healer's ST and Aoe heals get combined, further reducing button bloat for tank classes. 

This probably isn't super optimal but I've ran a few hundred extremes with this setup on a personal combo fork and have not had any issues with it. For casual content it is more than good enough for rolling mits.

WAR Examples:
- Rampart 
> Rampart changes to Damnnation > Thrill of Battle > Rampart based on whichever is avaliable & has the longest CD to get long CDs rolling earlier. Additionally, I've got Equilibrium on mine when below a certain HP threshold. If everything is on CD the icon shows whichever is coming back first. 

- Reprisal
> Reprisal changes to Shake It Off when there is no target, Reprisal is on CD or the target already has Reprisal with more than 2s remaining in the debuff.

I've experimented with including Bloodwhetting/Nascent Flash on the single target MIT button as well, but it fills a different niche mechanically so I personally left it out but I could see a use case for including it as well. Arm's Length is also not included as a mit for similar reasons. My other tanks are set up more or less the same way with their respective job actions.

Not at my PC currently, but I can provide the code I currently use as an example if that helps at all.

**Comment by zbee** [comment]
I think this is a pretty good idea.

After adding some basic mits and invulns to the main combos, like DRK in #33, I think I'll start working on putting this together.

Feel free to separate your code into it's own file and include it here as a file attachment, not a snippet, would be handy to have I'm sure.

**Comment by WesBosch** [comment]
Sure, it's based on ComboExpanded tho, not sloth, and has some... pretty ugly code in it lol. Would the full file still be useful? Or just the parts for mitigations lol. In either case, I'll toss some `// comments` in there to help explain my nonsense when I get home tonight

**Comment by zbee** [comment]
Separate the mitigation parts into their own file, and upload that, is what I mean, yeah :)

**Comment by WesBosch** [comment]
Feels a little wierd without the rest of the file, but this should cover what I had for mits. The only other differences in what I have VS sloth/wrath is that I include Raw Intuition in my DPS combo conditionally based on my HP/ally HP, Provoke is Shirk when targetting an ally, and Low Blow becomes Interject when any target nearby is interruptable. (Using a custom ReAction option to target them) The rest is more or less the same, if maybe a little less optimized than sloth lol

Additionally, I made a function specifically to compare cooldowns of multiple actions, so I included that as well and commented as much as I can. Hopefully this is useful!

Github doesn't like .cs files so it's a txt, but the formatting should still be valid for c#
[mits.txt](https://github.com/user-attachments/files/17809233/mits.txt)
